### PR TITLE
Iss76

### PR DESCRIPTION
**Pull request by MatakuT**

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -365,7 +369,19 @@ def update_files
                                 ' for assignment ' +
                                 "'#{@assignment.short_identifier}'.")
         end
-
+        # check if only required files are allowed for a submission
+        unless students_filename.length < 1 ||
+               required_files.length == 0 ||
+               @assignment.only_required_files == false
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: Just do !@assignment.only_required_files

File: app/helpers/assignments_helper.rb [hunk 2]
```diff
@@ -2,11 +2,18 @@ module AssignmentsHelper
...
-      assignment_file = render(partial: 'assignment_file', locals: {form: form, assignment_file: AssignmentFile.new})
+      assignment_file = render(partial: 'assignment_file',
+                               locals: { form: form,
+                                         assignment_file: AssignmentFile.new
+                                })
       page << %{
         var new_assignment_file_id = new Date().getTime();
         $('assignment_files').insert({bottom: "#{ escape_javascript assignment_file }".replace(/(attributes_\\d+|\[\\d+\])/g, new_assignment_file_id) });
         $('assignment_assignment_files_' + new_assignment_file_id + '_filename').focus();
+        if ($('assignment_only_required_files')){
```
Comment (david-yz-liu) [inline app/helpers/assignments_helper.rb]: Add a space before the `{`

File: app/views/assignments/_form.html.erb [hunk 3]
```diff
@@ -215,10 +216,30 @@
                              form: f } %>
       <% end %>
     </div>
-
      <p id='assignment_file_add_link'>
         <%= add_assignment_file_link t('assignment.add_required_file'), f %>
      </p>
+     <br>
+     <% if @show_checkbox_required_files == true %>
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: Remove the `== true`

File: app/views/assignments/_form.html.erb [hunk 4]
```diff
@@ -215,10 +216,30 @@
...
+	   <%= f.check_box :only_required_files,
+	                      checked: @assignment.only_required_files,
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: The alignment here is off.

File: app/views/assignments/_form.html.erb [hunk 5]
```diff
@@ -215,10 +216,30 @@
...
+	                      onclick: 'set_onbeforeunload(true);' 
+	   %>
+	   <%= f.label :only_required_files,
+	                 t('required_file_warning'),
+	                 { id: "assignment_only_required_files_label" }
+	   %>
+	 <% else %>
+	   <%= f.check_box :only_required_files,
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: This behaviour seems very redundant. I think the best thing to do is have the checkbox appear all the time.

File: config/locales/en.yml [hunk 6]
```diff
@@ -11,6 +11,9 @@ en:
     issues:             "GitHub issues list"
     actions:            "Actions"
 
+    # app/views/assignments/_form.html.erb
+    required_file_warning: "Accept no other files besides the listed required files. This would also restrics the file extensions."
```
Comment (david-yz-liu) [inline config/locales/en.yml]: Typo on 'restrics'

File: config/locales/en.yml [hunk 7]
```diff
@@ -150,6 +153,7 @@ en:
         add_required_file:  "Add a Required File"
         type:  "Assignment type"
         overall_comment: "Overall Comment"
+        upload_file_requirement: "The list of uploaded files does not match the assignment required files"
```
Comment (david-yz-liu) [inline config/locales/en.yml]: change last part to 'does not match the required files for this assignment.'

File: db/migrate/20150126155628_add_only_required_files_to_assignments.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddOnlyRequiredFilesToAssignments < ActiveRecord::Migration
+  def change
+    add_column :assignments, :only_required_files, :boolean, default: 'false'
```
Comment (david-yz-liu) [inline db/migrate/20150126155628_add_only_required_files_to_assignments.rb]: I don't think you need to put `false` in a string here.

File: app/views/assignments/_form.html.erb (hunk 5 above)
Comment (MatakuT) [inline app/views/assignments/_form.html.erb]: It is going to be much easier. But then when we want to create a new file the following sequence will appear:

Add filename(s) that students would be required to submit:
Add a Required File (a link)
(checkbox) Accept no other files besides the listed required files. This also restricts the file extensions.

which, I think, is not the most graceful way, as the appearance checkbox does not make sense until we see the form generated by clicking on "Add a Required File".

Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: This is true, but you have already spent more time on this feature than you probably should have. I'm fine closing it up with this.

Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: @MatakuT So can we make the checkbox always appear?

**Comment by david-yz-liu** [comment]
@MatakuT Okay, I merged this in, although either the checkbox should always appear, or it should disappear if all required files are removed.

**Pull request merged**